connector_search_engine/models/se_binding.py

```diff
@@ -122,92 +132,64 @@ def _jobify_get_job_description(self, processing_count):
 
-    def recompute_json(self, force_export=False):
-        try:
-            with self.env.cr.savepoint():
-                return self._recompute_json(force_export=force_export)
-        except Exception as e:
-            # If the batch fails, retry with a half len batch:
-            if len(self) > 1:
-                self.jobify_recompute_json(
-                    force_export=force_export,
-                    batch_size=math.ceil(len(self) / 2),
-                )
-                return _(
-                    "Job have been splited due to failling element.\nError: {}"
-                ) % str(e)
-            # We can't systematically reraise here, if we do the new jobs
-            # will be discarded.
-            raise
-
-    def _work_by_index(self, active=True):
-        self = self.exists()
-        for backend in self.mapped("backend_id"):
-            for index in self.mapped("index_id"):
-                bindings = self.filtered(
-                    lambda b, backend=backend, index=index: b.backend_id == backend
-                    and b.index_id == index
-                )
-                with backend.work_on(self._name, records=bindings, index=index) as work:
-                    yield work
-
-    def _recompute_json(self, force_export=False):
-        """Compute index record data as JSON."""
+    def recompute_json(self, force_export: bool = False):
+        """ "Compute index record data as JSON."""
         # `sudo` because the recomputation can be triggered from everywhere
         # (eg: an update of a product in the stock) and is not granted
         # that the user triggering it has access to all required records
         # (eg: se.backend or related records needed to compute index values).
         # All in all, this is safe because the index data should always
         # be the same no matter the access rights of the user triggering this.
 
-        index2serializer = {}
-        index2validator = {}
-        for index in self.index_id:
-            index2serializer[index] = index.get_serializer()
-            index2validator[index] = index.get_validator()
-
         for record in self.sudo():
-            if not record.record_id.exists():
+            if not record.record:
                 record.state = "to_delete"
                 _logger.error(
                     "There is something wrong, the record do not exists "
                     "flag the binding to be deleted"
                 )
                 continue
-
+            index = record.index_id
             # force the lang if needed
-            if record.index_id.lang_id:
-                record = record.with_context(lang=self.index_id.lang_id.code)
+            if index.lang_id:
+                record = record.with_context(lang=index.lang_id.code)
 
-            record.data = index2serializer[index].serialize(record.record_id)
+            old_data = record.data
+            record.data = index.model_serializer.serialize(record.record)
```

Comment:
Question: What is better ?
- keeping the code like this (so if the serialization fail the job fail)
- adding a try except and set a particular state on the se_binding ?